ipaserver/plugins/baseuser.py

```diff
@@ -694,3 +703,159 @@ class baseuser_remove_principal(LDAPRemoveAttribute):
     def pre_callback(self, ldap, dn, entry_attrs, attrs_list, *keys, **options):
         ensure_last_krbprincipalname(ldap, entry_attrs, *keys)
         return dn
+
+
+certmap_options = (
+    DNParam(
+        'issuer?',
+        cli_name='issuer',
+        label=_('Issuer'),
+        doc=_('Issuer of the certificate'),
+        flags=['virtual_attribute', 'no_create', 'no_update']
+    ),
+    DNParam(
+        'subject?',
+        cli_name='subject',
+        label=_('Subject'),
+        doc=_('Subject of the certificate'),
+        flags=['virtual_attribute', 'no_create', 'no_update']
+    ),
+    Bytes(
+        'usercertificate*', validate_certificate,
+        cli_name='certificate',
+        label=_('Certificate'),
+        doc=_('Base-64 encoded user certificate'),
+    ),
+)
+
+
+def _convert_to_x500(name):
+    """
+    Converts a (ipa) DN into a string representation following X500 order
+    """
+    if name:
+        name.reverse()
```

Comment:
I'm always scared by in-place methods. Do you count with that the object where the original name reference was pointing to, changed also outside the `_convert_to_x500` method?

maybe we should rather implement get_reverse() method that returns a new DN and keeps the original reference untouched